src/farmer.rs

```diff
@@ -458,13 +458,13 @@ impl Farmer {
 
     /// Stops farming, closes plots, and sends signal to the node
     pub async fn close(self) {
+        let mut maybe_cmd_sender = self.cmd_sender.lock().await;
+        let cmd_sender = match maybe_cmd_sender.take() {
```

Comment:
seems like a good place to use the new `let else` to me :) 
Feel free to ignore though, not a critical suggestion
```suggestion
        let Some(cmd_sender) = maybe_cmd_sender.take() else {
            return;
        }
```